File: java/triangle/src/main/java/Triangle.java
public class Triangle {

    private double side1, side2, side3;

    public Triangle(double side1, double side2, double side3) throws TriangleException {
        if(side1 + side2 <= side3 || side2 + side3 <= side1 || side3 + side1 <= side2) {
            throw new TriangleException();
        }
        this.side1 = side1;
        this.side2 = side2;
        this.side3 = side3;
    }

    public boolean isEquilateral() {
        if(side1 == side2 && side1 == side3) {
            return true;
        } else return false;
    }

    public boolean isIsosceles() {
        if((side1 == side2 && side1 == side3) ||
           (side1 == side2 && side1 != side3) || 
           (side1 == side3 && side1 != side2) || 
           (side2 == side3 && side2 != side1)) {
            return true;
        } else return false;
    }

    public boolean isScalene() {
        if(side1 != side2 && side1 != side3) {
            return true;
        } else return false;
    }

}

File: python/triangle/triangle.py
def is_equilateral(sides):
    if sides[0] + sides[1] <= sides[2] \
            or sides[2] + sides[1] <= sides[0] \
            or sides[0] + sides[2] <= sides[1]:
        return False
    elif sides[0] == sides[1] and sides[0] == sides[2]:
        return True
    else:
        return False


def is_isosceles(sides):
    if sides[0] + sides[1] <= sides[2] \
            or sides[2] + sides[1] <= sides[0] \
            or sides[0] + sides[2] <= sides[1]:
        return False
    elif sides[0] == sides[1] \
            and sides[0] == sides[2]:
        return True
    elif sides[0] == sides[1] \
            and not sides[0] == sides[2]:
        return True
    elif sides[0] == sides[2] \
            and not sides[0] == sides[1]:
        return True
    elif sides[2] == sides[1] \
            and not sides[2] == sides[0]:
        return True
    else:
        return False


def is_scalene(sides):
    if sides[0] + sides[1] <= sides[2] \
            or sides[2] + sides[1] <= sides[0] \
            or sides[0] + sides[2] <= sides[1]:
        return False
    elif not sides[0] == sides[1] \
            and not sides[0] == sides[2] \
            and not sides[2] == sides[1]:
        return True
    else:
        return False
